derive_builder_core/src/macro_options/darling_opts.rs

```diff
@@ -127,13 +127,44 @@ impl Visibility for BuildFn {
 
 /// Contents of the `field` meta in `builder` attributes.
 #[derive(Debug, Clone, Default, FromMeta)]
-pub struct FieldMeta {
+pub struct StructLevelFieldMeta {
     public: Flag,
     private: Flag,
     vis: Option<syn::Visibility>,
 }
 
-impl Visibility for FieldMeta {
+impl Visibility for StructLevelFieldMeta {
+    fn public(&self) -> &Flag {
+        &self.public
+    }
+
+    fn private(&self) -> &Flag {
+        &self.private
+    }
+
+    fn explicit(&self) -> Option<&syn::Visibility> {
+        self.vis.as_ref()
+    }
+}
+
+/// Contents of the `field` meta in `builder` attributes.
+//
+// TODO this struct and its Visibility impl duplicate code from StructLevelFieldMeta
+// Ideally they would be combined somehow, but without adversely affecting diagnostics
+// from darling, etc.
```

Comment:
```suggestion
/// Contents of the `field` meta in `builder` attributes at the field level.
///
/// This is a superset of the attributes permitted in `field` at the struct level.
```

The TODO is true, but it's no different than `StructLevelSetter`, which doesn't have any such comment. I've reopened TedDriggs/darling#146 to evaluate adding `flatten` there; maybe link this to that?